prov/util/src/util_shm.c

```diff
@@ -321,11 +347,19 @@ int smr_map_add(const struct fi_provider *prov, struct smr_map *map,
 
 void smr_map_del(struct smr_map *map, int id)
 {
+	struct dlist_entry *entry;
+
 	if (id >= SMR_MAX_PEERS || id < 0 ||
 	    map->peers[id].peer.addr == FI_ADDR_UNSPEC)
 		return;
 
-	munmap(map->peers[id].region, map->peers[id].region->total_size);
+	pthread_mutex_lock(&ep_list_lock);
+	entry = dlist_find_first_match(&ep_name_list, smr_match_name,
+				       map->peers[id].peer.name);
+	if (!entry)
+		munmap(map->peers[id].region, map->peers[id].region->total_size);
+	pthread_mutex_unlock(&ep_list_lock);
+
```

Comment:
@shefty Like this?

Reply:
I think?  You can drop the lock after the find call here, since you're not accessing entry.  : D

Reply:
So picky! Updated :)